pkg/validate/validate.go

```diff
@@ -104,6 +104,10 @@ func validateProperties(p *api.Properties, externalOnly bool) (errs []error) {
 	if p.PublicHostname != "" && !isValidHostname(p.PublicHostname) {
 		errs = append(errs, fmt.Errorf("invalid properties.publicHostname %q", p.PublicHostname))
 	}
+	// for private preview
+	if p.PublicHostname != "" && p.PublicHostname != p.FQDN {
```

Comment:
I think we should be failing validate if PublicHostname != "" here.
Then, everywhere where we use PublicHostname, we would need to reference a (new) helper function that returned FQDN if PublicHostname == "", otherwise PublicHostname.

Reply:
I believe PublicHostname is already being defaulted to FQDN if it is "" during the plugin.Generate phase  (selectDNSNames in pkg/config/util.go).